docs/tutorial/03-resources.md

```diff
@@ -19,581 +19,588 @@ socialImageTitle: Cadence Resources
 
-#### Calls the `hello()` function
+:::info[Action]
 
-Next, we call the `hello()` function and log the output.
+Open the transaction named `Load Hello`.
 
-```cadence
-log(helloResource.hello())
+:::
+
+It's empty!
+
+:::info[Action]
+
+On your own, stub out a transaction that imports `HelloWorldResource` and passes in an account reference with the `BorrowValue` authorization entitlement.
+
+:::
+
+You should end up with something like this:
+
+```cadence load_hello.cdc
+import HelloWorldResource from 0x06
+
+transaction {
+
+    prepare(acct: auth(BorrowValue) &Account) {
+        // TODO
+    }
+}
 ```
 
-#### Saves the resource back in the signer's account
+You just learned how to `borrow` a reference to a resource.  You could use an `if` statement to handle the possibility that the resource isn't there, but if you want to simply terminate execution, a common practice is to combine a `panic` statement with the [nil-coalescing operator (`??`)].
 
-Next, we use `save` again to put the object back in storage in the same spot:
+This operator executes the statement on the left side.  If that is `nil`, the right side is evaluated and returned.  In this case, the return is irrelevant, because we're going to cause a `panic` and terminate execution.
+
+:::info[Action]
+
+Create a variable with a reference to the `HelloAsset` resource stored in the user's account.  Panic if this resource is not found.
+
+:::
 
 ```cadence
-acct.storage.save(<-helloResource, to: /storage/HelloAssetTutorial)
+let helloAsset = acct.storage.borrow<&HelloWorldResource.HelloAsset>(from: /storage/HelloAssetTutorial)
+    ?? panic("The signer does not have the HelloAsset resource stored at /storage/HelloAssetTutorial. Run the `Create Hello` Transaction to store the resource")
 ```
 
 :::info[Action]
 
-Select account `0x06` as the only signer. Click the `Send` button to submit
-the transaction.
+Finally, `log` the return from a call to the `hello()` function.
 
 :::
 
-You should see something like this:
+:::warning
 
+Borrowing a reference does **not** allow you to move or destroy a resource, but it **does allow** you to mutate data inside that resource.
+
+:::
+
+Your transaction should be similar to:
+
+```cadence
+import HelloWorldResource from 0x06
+
+transaction {
+    prepare(acct: auth(BorrowValue, LoadValue, SaveValue) &Account) {
+        let helloAsset = acct.storage.borrow<&HelloWorldResource.HelloAsset>(from: /storage/HelloAssetTutorial)
+            ?? panic("The signer does not have the HelloAsset resource stored at /storage/HelloAssetTutorial. Run the `Create Hello` Transaction again to store the resource")
+
+        log(helloAsset.hello())
+    }
+}
 ```
-"Hello, World!"
-```
+
+In Cadence, we have the resources to leave very detailed error messages. Check out the error messages in the [Non-Fungible Token Contract] and [Generic NFT Transfer transaction] in the Flow NFT GitHub repo for examples of production error messages.
```

Comment:
They're linked!  Reference-style links.  Or did that not work for you?

Reply:
Yeah they weren't getting linked for me, hmmm...

Reply:
Not like wikipedia.  Clicking the link where it is on the page goes directly to the link in a new window.  It renders and behaves exactly the same as the inline style of links.  But when reviewing the raw file, it will jump you down to the actual link at the bottom if you click on it.

Doing it this way improves the readability of the raw document and you can reuse links, so you don't have to put them inline every time you want to link to something, which discourages people from linking as much as is appropriate, and you don't have to update 10 of them if the link changes.